### PR TITLE
Analytics/infrastructure

### PR DESCRIPTION
**Pull request by evbambly**
Initial Firebase Analytics onboarding.

1. Tagged every page_view change from the AppRouter
2. Tagged every button in the Burger Menu

### CODE REVIEW

File: packages/donor/src/Analytics.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { logEvent } from "firebase/analytics";
+import { firebaseAnalytics } from "./firebase/FirebaseInitializer";
+
+export function reportScreen(path: string) {
+  if (!firebaseAnalytics) {
+    return;
+  }
+  console.log("screen" + path);
```
Comment (yaron1m) [inline packages/donor/src/Analytics.tsx:8]: Please don't commit the console prints

File: packages/donor/src/Analytics.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    firebase_screen: path,
+    firebase_screen_class: "donor_screen",
+  });
+}
+
+export function reportClick(path: string, buttonName: string) {
+  if (!firebaseAnalytics) {
+    return;
+  }
+  console.log(`click${path}/${buttonName}`);
```
Comment (yaron1m) [inline packages/donor/src/Analytics.tsx:19]: Same

**Review by yaron1m — APPROVED** [review]

File: packages/donor/src/Analytics.tsx (hunk 1 above)
Comment (evbambly) [reply packages/donor/src/Analytics.tsx]: Done.

File: packages/donor/src/Analytics.tsx (hunk 2 above)
Comment (evbambly) [reply packages/donor/src/Analytics.tsx]: Done

**Pull request merged**